drawBot/ui/drawBotController.py

```diff
@@ -167,6 +167,19 @@ def checkSyntax(self, sender=None):
         self.stdout = None
         self.stderr = None
 
+    def formatCode(self, sender=None):
+        import black
+        # get the code
+        code = self.code()
+        # format the code with black
+        formattedCode = black.format_str(code, mode=black.Mode())
```

Comment:
`black.format_str()` can raise `black.InvalidInput`, for example if the source contains a syntax error.